python/tvm/_ffi/base.py

```diff
@@ -48,8 +48,11 @@ def _load_lib():
     """Load libary by searching possible path."""
     lib_path = libinfo.find_lib_path()
     lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_GLOBAL)
-    # DMatrix functions
     lib.TVMGetLastError.restype = ctypes.c_char_p
+    # Put the libpath to LD_LIBRARY_PATH
+    # will be useful for pipe session to find libtvm
+    os.environ["LD_LIBRARY_PATH"] = "%s:%s" % (
```

Comment:
is this just to make minrpcserver work? if so, could we just add in the search path in that binary rather than changing LD_LIBRARY_PATH for all library loads here? or at least, can you revert the change when you're done?

i think this means that all hooks that invoke a subprocess inherit the modified LD_LIBRARY_PATH

Reply:
Given that it makes sense to make libtvm available to subprocesses, I think we can leave it as it is. While setting rpath for the binary would work if the binary is on the same machine, it may not work after we send it through an RPC, so it might be helpful to keep the env variable.

Reply:
i'm worried that something else might be in the same directory as libtvm.so, which is where the subtle bug may come from. I don't think it's a good idea to muck with LD_LIBRARY_PATH unless you're being really explicit about this.

Reply:
Note that the change to LD_LIBRARY_PATH won't affert the execution after the python script exits.   I still think it makes sense to make the libary path in the installation to be available for sub-processes

Reply:
correct me if i'm wrong, doesn't this mean that any python script that imports tvm and then launches a subprocess will launch a subprocess with LD_LIBRARY_PATH set? we really don't want that. if the subprocess depends on any library that happens to be in the same dir as libtvm.so, it will wind up loading that instead of the system-installed copy. this is a really subtle bug that is very difficult to track down.